provisioner/src/lib.rs

```diff
@@ -478,6 +478,19 @@ impl Provisioner for ShuttleProvisioner {
         if !ProjectName::is_valid(&request.project_name) {
             return Err(Status::invalid_argument("invalid project name"));
         }
+
+        // Check project ownership.
+        if !claim
+            .owns_project(&self.gateway_client, &request.project_name)
+            .await
+            .map_err(|_| Status::internal("can not verify project ownership"))?
+        {
+            let msg = "the request lacks the authorizations";
+            let status = Status::permission_denied(msg);
+            error!(error = &status as &dyn std::error::Error, msg);
```

Comment:
Thanks! Could we write it a bit simpler, like the suggestion below?
```suggestion
            let status = Status::permission_denied("the request is unauthorized");
            error!(error = &status as &dyn std::error::Error);
```

Reply:
Perhaps a comment is good here too, since we wouldn't usually emit an error for a 4xx. We could mention that we emit an error event here because it indicates an attempt at something malicious.

Reply:
Done